src/main/java/decodex/modules/ModuleManager.java

```diff
@@ -1,51 +1,142 @@
 package decodex.modules;
 
+import java.util.logging.Logger;
+
 import decodex.Decodex;
+import decodex.data.exception.ModuleException;
 import decodex.data.exception.UnknownModuleException;
 import decodex.modules.base64.Base64Decoder;
 import decodex.modules.base64.Base64Encoder;
+import decodex.modules.binary.BinaryDecoder;
+import decodex.modules.binary.BinaryEncoder;
 import decodex.modules.hex.HexDecoder;
 import decodex.modules.hex.HexEncoder;
-import java.util.logging.Logger;
+import decodex.modules.rot.RotEncoder;
+import decodex.ui.messages.ErrorMessages;
 
 /**
  * ModuleManager maintains the list of available modules and returns selected modules.
  */
 public class ModuleManager {
 
-    private Logger logger = Decodex.logger;
+    private final Logger logger = Decodex.logger;
+
+    private static final int MODULE_NAME_INDEX = 0;
 
     /**
-     * Array of available modules.
+     * Available modules.
      */
-    private final Module[] modules = {new Base64Encoder(), new Base64Decoder(), new HexEncoder(), new HexDecoder()};
+    private static final String[][] MODULE_LIST = {
+            {Base64Encoder.MODULE_NAME, Base64Encoder.MODULE_DESCRIPTION},
+            {HexEncoder.MODULE_NAME, HexDecoder.MODULE_DESCRIPTION},
+            {BinaryEncoder.MODULE_NAME, BinaryDecoder.MODULE_DESCRIPTION},
+            {RotEncoder.MODULE_NAME, RotEncoder.MODULE_DESCRIPTION}
+    };
 
     public ModuleManager() {
         // Log available modules
-        for (Module module : modules) {
-            logger.fine(String.format("Loaded %s", module.getName()));
+        for (String[] module : MODULE_LIST) {
+            logger.fine(String.format("Loaded %s", module[MODULE_NAME_INDEX]));
         }
     }
 
-    public Module[] getModules() {
-        return modules;
+    public String[][] getModules() {
+        return MODULE_LIST;
     }
 
     /**
      * Select module from a provided module name.
```

Comment:
Maybe "selects" over "select" to standardize for our action words in the beginning of the javadoc sentence.